### PR TITLE
🌱 Bump Cluster API to v1.7.0-beta.1

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: templates/clusterclass-template.yaml [hunk 1]
```diff
@@ -201,13 +201,15 @@ spec:
           controlPlane: true
     name: kubeVipPodManifest
   variables:
-  - name: sshKey
+  - metadata: {}
```
Comment (sbueringer) [inline templates/clusterclass-template.yaml:204]: Looks a bit strange but it's the same for other places where we have ObjectMeta. The only way to change this is to make the metadata field a pointer. I think I wouldn't do that.

File: templates/clusterclass-template-supervisor.yaml [hunk 2]
```diff
@@ -187,13 +187,15 @@ spec:
           controlPlane: true
     name: kubeVipPodManifest
   variables:
-  - name: sshKey
+  - metadata: {}
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply templates/clusterclass-template-supervisor.yaml:190]: Just to answer here as well. Yes but in core CAPI obviously :)

**Comment by sbueringer** [comment]
We'll have to bump core CAPI images used in CI, otherwise it won't work with the schema change to ClusterClass

File: templates/clusterclass-template-supervisor.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (chrischdi) [reply templates/clusterclass-template-supervisor.yaml:190]: Yep that at least also breaks things:

> Error from server (BadRequest): error when creating "STDIN": ClusterClass in version "v1beta1" cannot be handled as a ClusterClass: strict decoding error: unknown field "spec.variables[0].metadata", unknown field "spec.variables[1].metadata", unknown field "spec.variables[2].metadata", unknown field "spec.variables[3].metadata", unknown field "spec.variables[4].metadata", unknown field "spec.variables[5].metadata"

[the remark this answers is not in the document]
Comment (sbueringer) [reply templates/clusterclass-template-supervisor.yaml:190]: Yeah but this is the other way around and that's fine

[the remark this answers is not in the document]
Comment (sbueringer) [reply templates/clusterclass-template-supervisor.yaml:190]: I'll look into the test errors when I'm back. Not sure why it's still failing

[the remark this answers is not in the document]
Comment (sbueringer) [reply templates/clusterclass-template-supervisor.yaml:190]: Tests still deploy 1.6.1. have to figure out how to do this properly. Maybe we have to use v1.7.0 in e2e config. So it's not handled as pre release

[the remark this answers is not in the document]
Comment (chrischdi) [reply templates/clusterclass-template-supervisor.yaml:190]: Can't we just say somewhere that its ok to use a prerelease?

[the remark this answers is not in the document]
Comment (sbueringer) [reply templates/clusterclass-template-supervisor.yaml:190]: Maybe. Feel free to take a look. Atm I think the problem is that the auto discovery picks latest non pre release. We could theoretically pin to the beta but this probably requires modifying a lot of tests

(Just not on my laptop right now)

File: test/e2e/config/vsphere.yaml [hunk 3]
```diff
@@ -34,13 +34,23 @@ providers:
   - name: cluster-api
     type: CoreProvider
     versions:
+      - name: v1.7.0-beta.1
```
Comment (chrischdi) [inline test/e2e/config/vsphere.yaml:37]:
```suggestion
      - name: v1.7.0
```

File: templates/clusterclass-template-supervisor.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (chrischdi) [reply templates/clusterclass-template-supervisor.yaml:190]: I think you're right, there's no way to get around this other than using the name `- name: v1.7.0` in our config.

[the remark this answers is not in the document]
Comment (chrischdi) [reply templates/clusterclass-template-supervisor.yaml:190]: I think you're right, there's no way to get around this other than using the name `- name: v1.7.0` in our config.

**Comment by sbueringer** [comment]
@fabriziopandini @chrischdi I think this PR should be ready now

**Review by chrischdi — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**